webpush/push.go

```diff
@@ -69,6 +69,42 @@ func NewPushRequest(sub *Subscription, message string, token string) (*http.Requ
 	return req, nil
 }
 
+// NewVapidRequest creates a valid Web Push HTTP request for sending a message
+// to a subscriber, using Vapid authentication. You can add more headers to
+// configure collapsing, TTL.
+func NewVapidRequest(sub *Subscription, message string, vapid *Vapid) (*http.Request, error) {
+	// If the endpoint is GCM then we temporarily need to rewrite it, as not all
+	// GCM servers support the Web Push protocol. This should go away in the
+	// future.
+	req, err := http.NewRequest("POST", sub.Endpoint, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	tok := vapid.Token(sub.Endpoint)
+	req.Header.Add("Authorization", fmt.Sprintf(`Bearer %s`, tok))
```

Comment:
The spec had Bearer few versions back, I'll check if the new name works with the servers.